src/Akka.Persistence.SqlServer/Snapshot/SqlServerSnapshotStore.cs

```diff
@@ -28,7 +28,7 @@ public SqlServerSnapshotStore(Config config) : base(config)
                 manifestColumnName: "Manifest",
                 timestampColumnName: "Timestamp",
                 serializerIdColumnName: "SerializerId",
-                timeout: sqlConfig.GetTimeSpan("connection-timeout"),
+                timeout: config.GetTimeSpan("connection-timeout"),
```

Comment:
I see the real issue was this line:

```
snapshotTableName: config.GetString("table-name"),
```

Before your change, `config` never had the fallback itself - that was applied, immutably, afterwards. This makes sense now.